nexus/src/app/affinity.rs

```diff
@@ -314,23 +307,28 @@ impl super::Nexus {
     ) -> Result<external::AffinityGroupMember, Error> {
         let (.., authz_affinity_group) =
             affinity_group_lookup.lookup_for(authz::Action::Modify).await?;
-        let (.., authz_instance) =
-            instance_lookup.lookup_for(authz::Action::Read).await?;
-        let member = external::AffinityGroupMember::Instance(
-            InstanceUuid::from_untyped_uuid(authz_instance.id()),
-        );
+        let (.., authz_instance, instance) =
+            instance_lookup.fetch_for(authz::Action::Read).await?;
+        let member = InstanceUuid::from_untyped_uuid(authz_instance.id());
 
         self.db_datastore
-            .affinity_group_member_add(
+            .affinity_group_member_instance_add(
                 opctx,
                 &authz_affinity_group,
-                member.clone(),
+                member,
             )
             .await?;
-        Ok(member)
+        Ok(external::AffinityGroupMember::Instance {
+            id: member,
+            name: instance.name().clone(),
+            // TODO: This is kinda a lie - the current implementation of
+            // "affinity_group_member_instance_add" relies on the instance
+            // note having a VMM, but that might change in the future.
```

Comment:
```suggestion
            // not having a VMM, but that might change in the future.
```

Same typo on line 356.

Reply:
Will update

Reply:
Done in 4e95b23f482eb218953a96817264200db13c0d1b